### PR TITLE
Add more detail to installation documentation

### PR DESCRIPTION
**Pull request by samuelpowell**
I will push commits here as I add (what I hope is) useful information to the installation documentation. @tanmaykm, please comment/review as necessary.

### CODE REVIEW

**Comment by samuelpowell** [comment]
@tanmaykm, I have reorganised the documentation of a single machine installation to contain an overview, then step-by-step instructions. I think this allows for some more details of what each step actually does, without confusing the actual instructions. That said, you may find it overly verbose! 

If you think this is useful (and accurate), let me know and I will squash.

File: docs/INSTALL.MD [hunk 1]
```diff
@@ -1,18 +1,90 @@
...
-    - `git clone https://github.com/JuliaLang/JuliaBox.git`
-2. Install docker and a few more packages required to develop and administer JuliaBox:
-    - Modify `JuliaBox/scripts/install/sys_install.sh` and set `DOCKER_FS` to `AUFS` instead of `DEVICEMAPPER`.
-        - This will avoid certain unnecessary configurations required only in certain production environments.
-    - Run `JuliaBox/scripts/install/sys_install.sh`
-3. Create work folders: `mkdir -p /jboxengine/conf /jboxengine/data /jboxengine/data/db`
-4. Configure JuliaBox:
+## Single-machine setup
+
+### Installation overview
```
Comment (tanmaykm) [inline docs/INSTALL.MD]: Overview applies to even the cluster setup. Should probably be right under the "Installation" section.

File: docs/INSTALL.MD [hunk 2]
```diff
@@ -1,18 +1,90 @@
...
+Installing JuliaBox on a single machine consists of five principle steps, the specifics of which vary depending upon the desired configuration. The following is a general overview of the process, and the key scripts and files involved. It may be useful to consult the included `Architecture.MD` document to gain a high-level overview of JuliaBox before reading these instructions.
```
Comment (tanmaykm) [inline docs/INSTALL.MD]: may be better to link references directly. i.e. [architecture document](Architecture.md)

File: docs/INSTALL.MD [hunk 3]
```diff
@@ -1,18 +1,90 @@
...
+	Finally, the necessary work folders to store data & configuration items must be created.
+	
+2. JuliaBox configuration.
+
+	The `jbox.user` JuliaBox configuration file is created according to templates provided later in this document. This file sets:
+	- limits on the maximum number of active containers, and disks,
+	- details of administrators, and,
+	- the plugins to be used for this deployment.
+
+	The `jbox_configure.sh` script is executed to configure the JuliaBox session ID key, and move the server configuration templates to the work directories (from where they will be mounted by the Docker containers).
```
Comment (tanmaykm) [inline docs/INSTALL.MD]: The `jbox_configure.sh` script is executed to configure the JuliaBox installation secret (used as part of encryption key for signing and encrypting session data), and move...

File: docs/INSTALL.MD [hunk 4]
```diff
@@ -28,34 +100,54 @@ JuliaBox is developed and tested on Ubuntu 14.04. But it should be easy to adapt
...
+			- [Review available images](https://github.com/tanmaykm/JuliaDockerImages)
+			- Pull selected image, e.g., `docker pull julialang/juliabox:v0.3.11`
+			
+		- To build an image:
+			- Run `docker pull ubuntu:14.04`.
+			- Clone template Dockerfiles: `git clone https://github.com/tanmaykm/JuliaDockerImages.git`.
+			- Follow instructions in repository to build and tag images.
+	
+	- Build JuliaBox images:
+		- Modify `FROM` in `JuliaBox/docker/Dockerfile` to point to the correct base image.
```
Comment (tanmaykm) [inline docs/INSTALL.MD]: Should also probably mention how to change the bundled default packages:

- If the chosen image is not the default JuliaBox image (`julialang/juliaboxpkgdist`), you may also remove undesired packages from `JuliaBox/docker/setup_julia.sh` or add new ones of your choice.

**Comment by tanmaykm** [comment]
@samuelpowell Thanks, this looks great! I have added some comments. Please squash the commits whenever you feel they are ready.

**Comment by samuelpowell** [comment]
@tanmaykm, thanks for the review. Updated and squashed. Let me know if you require any more changes.

**Comment by tanmaykm** [comment]
Thanks!

**Pull request merged**